specs/simple-serialize.md

```diff
@@ -1,424 +1,141 @@
-   + [Signed Roots](#signed-roots)
-* [Implementations](#implementations)
-
-## About
-
-`SimpleSerialize` was first proposed by Vitalik Buterin as the serialization
-protocol for use in the Ethereum 2.0 Beacon Chain.
-
-The core feature of `ssz` is the simplicity of the serialization with low
-overhead.
-
-## Variables and Functions
-
-| Term         | Definition                                                                                     |
-|:-------------|:-----------------------------------------------------------------------------------------------|
-| `little`     | Little endian.                                                                                 |
-| `byteorder`  | Specifies [endianness](https://en.wikipedia.org/wiki/Endianness): big endian or little endian. |
-| `len`        | Length/number of bytes.                                                                        |
-| `to_bytes`   | Convert to bytes. Should take parameters ``size`` and ``byteorder``.                           |
-| `from_bytes` | Convert from bytes to object. Should take ``bytes`` and ``byteorder``.                         |
-| `value`      | The value to serialize.                                                                        |
-| `rawbytes`   | Raw serialized bytes.                                                                          |
-| `deserialized_object` | The deserialized data in the data structure of your programming language.             |
-| `new_index`  | An index to keep track the latest position where the `rawbytes` have been deserialized.        |
+# SimpleSerialiZe (SSZ)
+
+This is a **work in progress** describing typing, serialization and Merkleization of Ethereum 2.0 objects.
+
+## Table of contents
+
+- [Constants](#constants)
+- [Typing](#typing)
+    - [Basic types](#basic-types)
+    - [Composite types](#composite-types)
+    - [Notation](#notation)
+    - [Aliases](#aliases)
+- [Serialization](#serialization)
+    - [`uintN`](#uintn)
+    - [`bool`](#bool)
+    - [Containers](#containers)
+    - [Tuples](#tuples)
+    - [Lists](#lists)
+- [Deserialization](#deserialization)
+- [Merkleization](#merkleization)
+- [Self-signed containers](#self-signed-containers)
+- [Implementations](#implementations)
 
 ## Constants
 
-| Constant          | Value | Definition                                                                            |
-|:------------------|:-----:|:--------------------------------------------------------------------------------------|
-| `LENGTH_BYTES`    |   4   | Number of bytes used for the length added before a variable-length serialized object. |
-| `SSZ_CHUNK_SIZE`  |  128  | Number of bytes for the chunk size of the Merkle tree leaf.                           |
+| Name | Value | Description |
+|-|-|-|
+| `LENGTH_BYTES` | `4` | Number of bytes for the length of variable-length serialized objects. |
+| `MAX_LENGTH` | `2**(8 * LENGTH_BYTES)` | Maximum serialization length. |
```

Comment:
Should the chunk size be defined here as well (with a note that it must be greater or equal than the maximum size of a basic type)

Reply:
The chunk size is chosen to match the output size of `hash` so is not really configurable.